### PR TITLE
reduce compile times by specializing broadcasts to loops

### PR DESCRIPTION
**Pull request by ChrisRackauckas**
Companion PR to https://github.com/SciML/OrdinaryDiffEq.jl/pull/1465

### CODE REVIEW

File: src/calculate_residuals.jl [hunk 1]
```diff
@@ -91,6 +91,12 @@ in `out`.
   nothing
 end
 
+@inline function calculate_residuals!(out::Array, ũ::Array, u₀::Array, u₁::Array, α::Number, ρ::Number, internalnorm,t)
+  for i in eachindex(u₀)
```
Comment (YingboMa) [inline src/calculate_residuals.jl:95]:
```suggestion
  @inbounds for i in eachindex(out, ũ, u₀, u₁)
```

File: src/calculate_residuals.jl [hunk 2]
```diff
@@ -91,6 +91,12 @@ in `out`.
   nothing
 end
 
+@inline function calculate_residuals!(out::Array, ũ::Array, u₀::Array, u₁::Array, α::Number, ρ::Number, internalnorm,t)
```
Comment (YingboMa) [inline src/calculate_residuals.jl:94]:
```suggestion
@inline function calculate_residuals!(out::Array, ũ::Array, u₀::Array, u₁::Array, α::Number, ρ::Number, internalnorm::F, t) where F
```

File: src/calculate_residuals.jl [hunk 3]
```diff
@@ -91,6 +91,12 @@ in `out`.
...
+  for i in eachindex(u₀)
+    out[i] = calculate_residuals(ũ[i], u₀[i], u₁[i], α, ρ, internalnorm,t)
```
Comment (YingboMa) [inline src/calculate_residuals.jl:96]:
```suggestion
    out[i] = calculate_residuals(ũ[i], u₀[i], u₁[i], α, ρ, internalnorm, t)
```

Comment (chriselrod) [reply src/calculate_residuals.jl]: No `@inbounds`?

**Pull request merged**